### DISCUSSION TITLE
Alter default `IsolationMode` to `InstancePerTest` instead of `InstancePerSpec`

### DISCUSSION THREAD

**Discussion by LeoColman** [kotest/kotest#3944] [6.0 changes]
I know we discussed this in 3.0, 4.0, 5.0 and now again 😂 

I believe this is an opportunity to replace the default `isolationMode` from `InstancePerSpec` to `InstancePerTest`. It's more in-line with what new users expect when coming from JUnit (majority of kotest new users)

**Comment by Kantis** [comment]
I'll add a vote against this, because I feel switching defaults this far into the project would be a nuisance to existing users. I haven't really got the impression that anyone is bothered by this default. Maybe the votes on this discussion will prove me wrong though 👀

**Reply by LeoColman** [reply]
I'm pretty bothered by the default because I always forget about it and have to add `InstancePerTest` to all tests that depends on automatic cleanup.

Even after using Kotest since KotlinTest I still forget about it, that's why I feel this change deserves some love

**Reply by sksamuel** [reply]
In theory it might be better but I agree with @Kantis we're far too down the line to switch the default now.
Plus you can change it globally with config right.
